src/js/admin/screens/groupsScreen.js

```diff
@@ -0,0 +1,150 @@
+      <div class="cp-control">${groupKeys.length}&nbsp;<i class="material-icons">storage</i></div>
+    </div>
+    `);
+    App.jControlPanelHeader.replaceWith(header);
+    App.jControlPanelHeader = header;
+    const cpBody = $(`
+    <div>
+        <div class="card-header d-flex justify-content-between align-items-center">
+            <h5>Manage groups</h5>
+            <button class="btn btn-primary"><i class="material-icons">import_export</i>&nbsp;Import/Export</button>
+        </div>
+        <div class="card-body">
+        <p class="card-text">Add new group by entering its key</p>
+        <form id="group-addition">
+          <div class="input-group">
+            <input class="form-control invalid" placeholder="Enter new groups’ key" title="Key 1-20 digits">
+            <button class="btn btn-primary btn-raised"><i class="material-icons">add</i>&nbsp;Add new</button>
+          </div>
+        </form>
+      </div>
+        <div id="groups-table" class="table"></div>
+    </div>
+    `);
+    App.jGroupsContainer = cpBody.find('#groups-table');
+    const form = cpBody.find('#group-addition');
+    const input = form.find('input');
+    
+    input.keyup(function(){
+        const newKey = input.val();
+        if(newKey && !isNaN(newKey) && !groupKeys[parseInt(newKey)]) {
+            input.removeClass("invalid");
+        } else {
+            input.addClass("invalid");
+        }
+    });
+    form.submit((e) => {
+        e.preventDefault();
+        const newKey = input.val();
+        if (newKey && !isNaN(newKey) && !groupKeys[newKey]) {
+          App.showGroupEditForm(newKey);
+        }
+    });
+    
+    App.renderGroupsTable();
+    App.jControlPanelBody.replaceWith(cpBody);
+    App.jControlPanelBody = cpBody;
+}
+
+App.renderGroupsTable = () => {
+    const groupKeys = Object.keys(App.groups);
+    groupKeys.sort((a, b) => App.groups[a]["number"] - App.groups[b]["number"]); // Descending order
+
+    App.jGroupsContainer.empty();
+    App.jGroupsContainer.append(tableHeader);
+    for (let i = 0; i < groupKeys.length; i++) {
+        const groupKey = groupKeys[i];
+        const group = App.groups[groupKey];
+        const { name, number, img } = group || {};
+        const style = ` style="background-image: url(${img})"`;
+        const item = $(`
+            <div class="tr search-result">
+                <div class="td sr-img" ${style}></div>
+                <div class="td sr-number">${number}</div>
+                <div class="td sr-id">${groupKeys[i]}</div>
+                <div class="td sr-name">${name}</div>
+                <button class="td sr-edit btn btn-primary"><i class="material-icons">edit</i></button>
+            </div>
+        `);
+        item.children('.sr-edit, .sr-name').click(() => {
+            App.showGroupEditForm(groupKey, () => input.keyup());
+        });
+        App.jGroupsContainer.append(item);
+    }
+
+    App.showGroupEditForm = (key, cb) => {
```

Comment:
tuhle funkci myslim ze nebudeme vojat jinde nez v tomhle souboru, takze udelej z ni lokalni, tj. nedavat to do globalniho objektu `App`

Reply:
Takže to samé pro funkci v productScreen?

Reply:
v App.js je zakomentovaný
//App.connect().then(() => {
  //  App.showProductEditForm('6333');
//})

Reply:
jo to jsem schvalne udělal abych nemusel proklikat k tomu formulari, aby se to hned zobrazilo po refresh stranky